src/python/pants/backend/python/rules/run_setup_py.py

```diff
@@ -194,44 +199,69 @@ def register_options(cls, register):
            "`--setup-py2-args=\"bdist_wheel --python-tag py36.py37\"`. If unspecified, we just "
            "dump the setup.py chroot."
     )
+    register(
+      '--recursive', type=bool,
+      help="If specified, will run the setup.py command recursively on all exported targets that "
+           "the specified targets depend on, in dependency order.  This is useful, e.g., when "
+           "the command publishes dists, to ensure that any dependencies of a dist are published "
+           "before it."
```

Comment:
Thank you for expanding this from the original V1. However, I'm confused with the references to publishing given this comment: https://github.com/pantsbuild/pants/issues/8912#issuecomment-572388758

Perhaps you mean `when building distributions in preparation for publishing`?

Reply:
Ah yes, that was crufty. Order no longer matters at all. Plus, I didn't actually implement this behavior... Fixed.